In [1]:
import sys
sys.path.append(r'C:\Users\Lenovo\Documents\IC')
import numpy as np

In [115]:
filename = 'all_solutions.npz'
data = np.load(filename)
gurobi_values = data["all_solutions"]
print(gurobi_values)

gurobi_10 = gurobi_values[0]
gurobi_30 = gurobi_values[1]
gurobi_100 = gurobi_values[2]

[ 7316203.9254538  19978014.71339107 70598781.54841062]


In [116]:
def percentage_difference(value1, value2):
    if value1 == value2:
        print("Both values are equal.")
        return
        
    difference = abs(value1 - value2)
    average = (value1 + value2) / 2
    percentage_diff = (difference / average) * 100
        
    if value1 > value2:
        print(f"{value1} is {percentage_diff:.2f}% greater than {value2}.")
    else:
        print(f"{value1} is {percentage_diff:.2f}% less than {value2}.")

# Problem size 10

## set 1

In [117]:
filename = '../sets1/FX_values_0.npz'
data = np.load(filename)
data = data["FX_values_"]

a = data[0, :]
b = data[1, :]

mean_a = np.mean(a)
mean_b = np.mean(b)

print(mean_a)
print(mean_b)

7471636.9085166175
7429329.95105469


In [118]:
percentage_difference(mean_a, gurobi_10)
percentage_difference(mean_b, gurobi_10)

7471636.9085166175 is 2.10% greater than 7316203.925453799.
7429329.95105469 is 1.53% greater than 7316203.925453799.


## set 2

In [119]:
filename = '../sets2/FX_values_0.npz'
data = np.load(filename)
data = data["FX_values_"]

d = data[3, :]

mean_d = np.mean(d)

print(mean_d)

7461814.234331965


In [120]:
percentage_difference(mean_d, gurobi_10)

7461814.234331965 is 1.97% greater than 7316203.925453799.


## set 3

In [121]:
filename = '../sets3/FX_values_0.npz'
data = np.load(filename)
data = data["FX_values_"]

a = data[0, :]
b = data[1, :]

mean_a = np.mean(a)
mean_b = np.mean(b)

print(mean_a)
print(mean_b)

7523831.679344869
7520207.929617735


In [122]:
percentage_difference(mean_a, gurobi_10)
percentage_difference(mean_b, gurobi_10)

7523831.679344869 is 2.80% greater than 7316203.925453799.
7520207.929617735 is 2.75% greater than 7316203.925453799.


# Problem size 30

## set 1

In [123]:
filename = '../problem_30/FX_values_0.npz'
data = np.load(filename)
data = data["FX_values_"]

a = data[0, :]
b = data[1, :]

mean_a = np.mean(a)
mean_b = np.mean(b)

print(mean_a)
print(mean_b)

20303925.458620314
20177253.46178516


In [124]:
percentage_difference(mean_a, gurobi_30)
percentage_difference(mean_b, gurobi_30)

20303925.458620314 is 1.62% greater than 19978014.71339107.
20177253.46178516 is 0.99% greater than 19978014.71339107.


## set 2

In [125]:
filename = '../problem_30/FX_values_2_0.npz'
data = np.load(filename)
data = data["FX_values_"]

d = data[3, :]

mean_d = np.mean(d)

print(mean_d)

20282736.07480412


In [126]:
percentage_difference(mean_d, gurobi_30)

20282736.07480412 is 1.51% greater than 19978014.71339107.


## set 3

In [127]:
filename = '../problem_30/FX_values_3_0.npz'
data = np.load(filename)
data = data["FX_values_"]

a = data[0, :]
b = data[1, :]

mean_a = np.mean(a)
mean_b = np.mean(b)

print(mean_a)
print(mean_b)

21501467.913612943
21174709.87288488


In [128]:
percentage_difference(mean_a, gurobi_30)
percentage_difference(mean_b, gurobi_30)

21501467.913612943 is 7.35% greater than 19978014.71339107.
21174709.87288488 is 5.82% greater than 19978014.71339107.


# Problem size 100

## set 1

In [129]:
filename = '../problem_100/FX_values_0.npz'
data = np.load(filename)
data = data["FX_values_"]

a = data[0, :]
b = data[1, :]

mean_a = np.mean(a)
mean_b = np.mean(b)

print(mean_a)
print(mean_b)

77131750.58050013
76202168.63172564


In [130]:
percentage_difference(mean_a, gurobi_100)
percentage_difference(mean_b, gurobi_100)

77131750.58050013 is 8.84% greater than 70598781.54841062.
76202168.63172564 is 7.63% greater than 70598781.54841062.


## set 2

In [131]:
filename = '../problem_100/FX_values_2_0.npz'
data = np.load(filename)
data = data["FX_values_"]

d = data[3, :]

mean_d = np.mean(d)

print(mean_d)

77212623.75365685


In [132]:
percentage_difference(mean_d, gurobi_100)

77212623.75365685 is 8.95% greater than 70598781.54841062.


## set 3

In [133]:
filename = '../problem_100/FX_values_3_0.npz'
data = np.load(filename)
data = data["FX_values_"]

a = data[0, :]
b = data[1, :]

mean_a = np.mean(a)
mean_b = np.mean(b)

print(mean_a)
print(mean_b)

83033737.83234812
83628406.09808718


In [134]:
percentage_difference(mean_a, gurobi_100)
percentage_difference(mean_b, gurobi_100)

83033737.83234812 is 16.19% greater than 70598781.54841062.
83628406.09808718 is 16.90% greater than 70598781.54841062.


## Size 10

### Set 1:
2.10% greater

1.53% greater

### Set 2:
1.97% greater

### Set 3:
2.80% greater

2.75% greater

## Size 30:

### Set 1:
1.62% greater

0.99% greater

### Set 2:
1.51% greater

### Set 3:
7.35% greater

5.82% greater 

## Size 100:

### Set 1:
8.84% greater

7.63% greater

### Set 2:
8.95% greater

### Set 3:
16.19% greater 

16.90% greater